### PR TITLE
chore(client): updated pipeline to godot version 4.2.2

### PR DESCRIPTION
**Pull request by Dracks**
* The project of godot doesn't seem to contain the patch version of godot. 
* The game seems to work fine, there was one unit that didn't had the color correctly during my match, but only one specific infantery. (I forget an screenshot)

### CODE REVIEW

**Comment by HunterNN** [comment]
I think we can remove the pre run for creating the import, since it is now included in the export flag:
https://godotengine.org/article/maintenance-release-godot-4-2-2-and-4-1-4/#improved-command-line-export-pipeline

**Comment by Dracks** [comment]
> I think we can remove the pre run for creating the import, since it is now included in the export flag: https://godotengine.org/article/maintenance-release-godot-4-2-2-and-4-1-4/#improved-command-line-export-pipeline

Super! I will try now to modify them :)

File: .github/workflows/client-type-check.yaml [hunk 1]
```diff
@@ -33,9 +33,8 @@ jobs:
         run: |
           cd client
           set +e
-          godot --project . --headless --quit-after 100 -e 2>/dev/null
-          godot --project . --headless --quit-after 100 -e 2>/dev/null
-          godot --project . --headless --quit-after 100 -e 2>/dev/null
+          godot --project . --headless --import -e 2>/dev/null
+          godot --project . --headless --import -e 2>/dev/null
```
Comment (Dracks) [inline .github/workflows/client-type-check.yaml:37]: it improved, but still we need to run it twice, I think the problem is that error we get on the first run allways. :(

File: .github/actions/setup-client-for-export/action.yml [hunk 2]
```diff
@@ -9,18 +9,36 @@ runs:
...
+
+        VERSION=${GITHUB_REF_NAME}
+        # In the case it's not a new tag, we use the commit sha
+        if [ "$GITHUB_REF_TYPE" == 'branch' ]; then
+          VERSION=${COMMIT_SHA:0:7}
+        fi
         sed -i "s|$VERSION_REGEX|$VERSION|g" client/levels/menu.tscn # Replace the literal string '<VERSION>' with the new constant declaration
-        sed -i "s|$VERSION_REGEX|$VERSION|g" client/project.godot # Replace the literal string '<VERSION>' with the new constant declaration
\ No newline at end of file
+        sed -i "s|$VERSION_REGEX|$VERSION|g" client/project.godot # Replace the literal string '<VERSION>' with the new constant declaration
```
Comment (Dracks) [inline .github/actions/setup-client-for-export/action.yml:35]: we can use the working-dir to not need the client/ path.

File: .github/workflows/client-type-check.yaml (hunk 1 above)
Comment (HunterNN) [reply .github/workflows/client-type-check.yaml:37]: Could you create an issue for that problem and reference this workaround (so it doesn't get forgotten)?

**Review by HunterNN — APPROVED** [review]

**Pull request merged**